constraint-solver/src/quadratic_symbolic_expression.rs

```diff
@@ -202,7 +204,7 @@ impl<T: FieldElement, V: Ord + Clone + Eq + Hash> QuadraticSymbolicExpression<T,
             // any symbolic expression.
             // We replace the variable by a symbolic expression, so it goes into the constant part.
             let coeff = self.linear.remove(variable).unwrap();
-            self.constant += &coeff * substitution;
+            self.constant += coeff * substitution.clone();
```

Comment:
I had to add a few clones, because `FieldElement` doesn't implement being able to multiply references, for example.

Now, what happens is that:
- We clone
- The implementation of the operator in `SymbolicExpression` delegates to the implementation on the reference
- This implementation clones again internally

I think the solution is for `SymbolicExpression` to implement the operators directly on the owned type.